node/node.go

```diff
@@ -482,8 +482,17 @@ func (n *Node) GetESDTsRoles(address string) (map[string][]string, error) {
 }
 
 // GetTokenSupply returns the provided token supply from current shard
-func (n *Node) GetTokenSupply(token string) (string, error) {
-	return n.processComponents.HistoryRepository().GetESDTSupply(token)
+func (n *Node) GetTokenSupply(token string) (*api.ESDTSupply, error) {
+	esdtSupply, err := n.processComponents.HistoryRepository().GetESDTSupply(token)
+	if err != nil {
+		return nil, err
+	}
+
+	return &api.ESDTSupply{
+		Supply: esdtSupply.Supply.String(),
```

Comment:
this works because how it is implemented right now. Since we are talking about 2 different components, I would put some nil checks for all fields (someone might change the behavior of the `HistoryRepository().GetESDTSupply` implementation in the future and the node will simply panic)

Reply:
`esdtSupply.Supply.String() `
if esdtSupply.Supply is nil will not panic.

Reply:
Right and we will index `<nil>` which, of course is a correctly formatted number :man_facepalming: 
Please add nil checks,

Reply:
added and unit tests